File: Sources/NSPopover-Extension.m
#import "NSPopover-Extension.h"

@implementation NSPopover (Extension)
+ (instancetype)popoverWithContentViewController:(NSViewController*)controller
{
    NSPopover *popover = [NSPopover new];
    popover.contentViewController = controller;
    popover.behavior = NSPopoverBehaviorTransient;
    return popover;
}

- (void)showAtRightOfButton:(NSButton*)button
                     ofView:(NSView*)view
                    offsetX:(CGFloat)x
                          Y:(CGFloat)y
{
    NSRect rect = button.frame;
    rect = NSMakeRect(rect.origin.x + x, rect.origin.y + y, rect.size.width, rect.size.height);
    
    [self showRelativeToRect:rect ofView:view preferredEdge:NSMaxXEdge];
}

+ (void)showPopoverWithViewController:(NSViewController*)controller
                      atRightOfButton:(NSButton*)button
                               ofView:(NSView*)view
                              offsetX:(CGFloat)x
                                    Y:(CGFloat)y
{
    [[NSPopover popoverWithContentViewController:controller] showAtRightOfButton:button
                                                                          ofView:view
                                                                         offsetX:x
                                                                               Y:y];
}

@end

File: Sources/NSPopover-Extension.swift
import Cocoa

extension NSPopover {
    convenience init(contentViewController controller: NSViewController) {
        self.init()
        self.contentViewController = controller
        self.behavior = .transient
    }

    func show(atRightOf button: NSButton, view: NSView, offsetX x: CGFloat, y: CGFloat) {
        let rect = button.frame
        let newRect = NSRect(x: rect.origin.x + x,
                             y: rect.origin.y + y,
                             width: rect.size.width,
                             height: rect.size.height)

        show(relativeTo: newRect, of: view, preferredEdge: .maxX)
    }

    @objc class func show(with controller: NSViewController,
                          atRightOf button: NSButton,
                          view: NSView,
                          offsetX x: CGFloat,
                          y: CGFloat) {
        let popover = NSPopover(contentViewController: controller)
        popover.show(atRightOf: button, view: view, offsetX: x,y: y)
    }
}
